pyat/at/physics/orbit.py

```diff
@@ -281,65 +315,78 @@ def find_sync_orbit(ring: Lattice, dct: float = None, refpts: Refpts = None, *,
+            "dp, dct and df may be specified"
+        )
     if orbit is None:
         if df is not None:
             frf = ring.cell_revolution_frequency * ring.cell_harmnumber
-            dct = -ring.cell_length * df / (frf+df)
+            dct = -ring.cell_length * df / (frf + df)
             orbit = _orbit_dct(ring, dct, keep_lattice=keep_lattice, **kwargs)
         elif dp is not None:
             orbit = _orbit_dp(ring, dp, keep_lattice=keep_lattice, **kwargs)
         else:
-            orbit = _orbit_dct(ring, dct,  keep_lattice=keep_lattice, **kwargs)
+            orbit = _orbit_dct(ring, dct, keep_lattice=keep_lattice, **kwargs)
         keep_lattice = True
 
     # bug in numpy < 1.13
     if ring.refcount(refpts) == 0:
-        all_points = numpy.empty((0, 6), dtype=float)
+        all_points = np.empty((0, 6), dtype=float)
     else:
-        all_points = internal_lpass(ring, orbit.copy(order='K'), refpts=refpts,
-                                    keep_lattice=keep_lattice)
-        all_points = numpy.squeeze(all_points, axis=(1, 3)).T
+        all_points = internal_lpass(
+            ring, orbit.copy(order="K"), refpts=refpts, keep_lattice=keep_lattice
+        )
+        all_points = np.squeeze(all_points, axis=(1, 3)).T
     return orbit, all_points
 
 
-def _orbit6(ring: Lattice, cavpts=None, guess=None, keep_lattice=False,
-            **kwargs):
+def _orbit6(ring: Lattice, cavpts=None, guess=None, keep_lattice=False, **kwargs):
     """Solver for 6D motion"""
-    convergence = kwargs.pop('convergence', DConstant.OrbConvergence)
-    max_iterations = kwargs.pop('max_iterations', DConstant.OrbMaxIter)
-    xy_step = kwargs.pop('XYStep', DConstant.XYStep)
-    dp_step = kwargs.pop('DPStep', DConstant.DPStep)
-    method = kwargs.pop('method', ELossMethod.TRACKING)
+    convergence = kwargs.pop("convergence", DConstant.OrbConvergence)
+    max_iterations = kwargs.pop("max_iterations", DConstant.OrbMaxIter)
+    xy_step = kwargs.pop("XYStep", DConstant.XYStep)
+    dp_step = kwargs.pop("DPStep", DConstant.DPStep)
+    method = kwargs.pop("method", ELossMethod.TRACKING)
     rem = kwargs.keys()
     if len(rem) > 0:
         raise AtError(f'Unexpected keywords for orbit6: {", ".join(rem)}')
 
     l0 = get_s_pos(ring, len(ring))[0]
     f_rf = ring.get_rf_frequency()
-    harm_number = round(f_rf*l0/ring.beta/clight)
+    harm_number = round(f_rf * l0 / ring.beta / clight)
 
     if guess is None:
-        _, dt = get_timelag_fromU0(ring, method=method, cavpts=cavpts)
+        try:
+            _, dt = get_timelag_fromU0(ring, method=method, cavpts=cavpts)
+        except AtError:
+            try:
+                _, dt = get_timelag_fromU0(
+                    ring, method=ELossMethod.INTEGRAL, cavpts=cavpts
+                )
+            except AtError:
+                msg = "Energy loss per turn calculation failed, closed orbit not found"
+                raise AtError(msg)
```

Comment:
I think that the check, if necessary, should be be done earlier in `get_timelag_fromU0`: if the energy loss computation fails for missing RF voltage, no need to try INTEGRAL, the tracking is right because it includes elements ignored by INTEGRAL, and the error message is correct. The problem might be if the tracking gives a NaN loss, and then I agree, one could try INTEGRAL. For this we need to check for NaN before the comparison with RF voltage.

So here, the option would be to catch the "timelag" error, complement the message to mention that it happens during closed orbit guess, and re-raise it.

However, I still do not see how we can get a NaN loss… We would need a dump of a failing lattice to examine it